docs/wiki/ModulesDesign.md

```diff
@@ -537,3 +538,45 @@ Parameter files in CARML leverage the common `deploymentParameters.json` schema
 - A module should have as many parameter files as it needs to evaluate all parts of the module's functionality.
 - Sensitive data should not be stored inside the parameter file but rather be injected by the use of [tokens](./ParameterFileTokens.md) or via a [key vault reference](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/key-vault-parameter?tabs=azure-cli#reference-secrets-with-static-id).
+
+# Resource Removal
+
+Removal step is triggered after the deployment completes. This is used for several reasons:
+- Make sure to keep the validation subscription cost as low as possible.
+- Enable testing of the full module deployment at every run.
+
+The default removal procedure works fine for most of the modules created so far, so it's likely you won't have to change anything to make your module to be removed correctly after deployment.
+
+## How it works
+
+The removal process will remove all resources created during deployment. The list is identified by:
+
+1. Getting the list of resources created by your deployment (even if created by a "nested" deployment - a deployment created by the parent one -)
+1. Ordering the list based on resourceId tokens length (ensures child resources are removed first)
+1. Removing from the list, if any, the resources used as dependencies for different modules (e.g. the commonly used Log Analytics workspace)
+1. If provided, putting on the top of the list of resources to remove specific resource types (as resource id length may not in all cases)
+
+After the removal of each resource, the script will attempt a **post removal operation**. This can be used for those resource types that requires a post processing, like purging a key vault.
+
+The procedure is initiated by the script `/utilities/pipelines/resourceRemoval/Initialize-DeploymentRemoval.ps1`, run during deployment by:
+- (Azure DevOps) `/.azuredevops/pipelineTemplates/module.jobs.deploy.yml`
+- (GitHub) `/.github/actions/templates/validateModuleDeployment/action.yml`
+
+It uses several helper scripts that can be found in `/utilities/pipelines/resourceRemoval`
+## Create a specialized removal procedure
+
+You can define a custom removal procedure by **modifying the resource types removal sequence** and by **defining a custom post-removal action**. The two are independent and you can act on one or both.
+
+To modify the resource types removal sequence:
+1. Open the `/utilities/pipelines/resourceRemoval/Initialize-DeploymentRemoval.ps1` file.
+1. Look for the following comment: `### CODE LOCATION: Add custom removal sequence here`
+1. Add a case value that matches your module name
+1. In the case block, update the `$removalSequence` variable value to accommodate your module requirements
+1. Remember to add the `break` statement.
+
```

Comment:
Add:

To add a custom removal step:
1. Open the `/utilities/pipelines/resourceRemoval/helper/Invoke-ResourceRemoval.ps1` file.
1. Look for the following comment: `### CODE LOCATION: Add custom removal sequence here`
1.  Add a case value that matches the resource type you want to add a custom removal operation for.

Reply:
@MrMCake  I totaly missed this method :) . Added it in my latest commit, please review